File: src/main/java/com/selqet/momrecipebook/entity/RecipeIngredientEntity.java
package com.selqet.momrecipebook.entity;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(schema = "public", name = "recipeingredient")
public class RecipeIngredientEntity {

    @EmbeddedId
    private RecipeIngredientKey id;

    @ManyToOne
    @MapsId("ingredientId")
    @JoinColumn(name = "ingr_id")
    private IngredientEntity ingredient;

    @ManyToOne
    @MapsId("recipeId")
    @JoinColumn(name = "rec_id")
    private RecipeEntity recipe;

    @Column(name = "amount")
    private Short amount;

    @Column(name = "unit")
    private String unit;
}

File: src/main/kotlin/com/example/momrecipebook/model/RecipeIngredientEntity.kt
package com.example.momrecipebook.model

import java.io.Serializable
import javax.persistence.*

@Embeddable
data class RecipeIngredientKey(
    @Column(name = "rec_id")
    var recipeId: Long,

    @Column(name = "ingr_id")
    var ingredientId: Long,
) : Serializable

@Entity
@Table(name = "recipeingredient")
data class RecipeIngredientEntity(
    @EmbeddedId
    var id: RecipeIngredientKey,

    @ManyToOne
    @MapsId("recipeId")
    @JoinColumn(name = "rec_id")
    var recipe: RecipeEntity,

    @ManyToOne
    @MapsId("ingredientId")
    @JoinColumn(name = "ingr_id")
    var ingredient: IngredientEntity,

    @Column(name = "amount")
    var amount: Short?,

    @Column(name = "unit")
    var unit: String?,
)